### PR TITLE
Support seconds, minutes, and hours in time_bucket_ng()

### PR DESCRIPTION
**Pull request by afiskon**
As a future replacement for time_bucket(), time_bucket_ng()
should support seconds, minutes, and hours. This patch adds
this support. The implementation is the same as for
time_bucket(). Timezones are not yet supported.

### CODE REVIEW

**Review by mfundul — APPROVED** [review]

File: src/time_bucket.c [hunk 1]
```diff
@@ -292,3 +292,168 @@ ts_time_bucket_by_type(int64 interval, int64 timestamp, Oid timestamp_type)
...
+		}
+
+		/*
+		 * Handle minutes, hours and days.
+		 */
+		Timestamp origin = (PG_NARGS() > 2 ? PG_GETARG_TIMESTAMP(2) : DEFAULT_ORIGIN);
+		Timestamp result;
+		int64 period = get_interval_period_timestamp_units(interval);
+
+		if (TIMESTAMP_NOT_FINITE(timestamp))
```
Comment (mkindahl) [inline src/time_bucket.c:320]: Codecov complains that there is no test for this case.

File: src/time_bucket.c [hunk 2]
```diff
@@ -292,3 +292,168 @@ ts_time_bucket_by_type(int64 interval, int64 timestamp, Oid timestamp_type)
...
+
+	/*
+	 * Discard any time information and work with the date.
+	 */
+	ts_date = DatumGetDateADT(DirectFunctionCall1(timestamp_date, PG_GETARG_DATUM(1)));
+
+	if (PG_NARGS() > 2)
+	{
+		DateADT result;
+		DateADT origin = DatumGetDateADT(DirectFunctionCall1(timestamp_date, PG_GETARG_DATUM(2)));
```
Comment (mkindahl) [inline src/time_bucket.c:336]: Do you have a test case for `NULL` here or is it not needed?

**Review by mkindahl — APPROVED** [review]
Just a question about tests, otherwise looks good, so approving. Please check if you need additional tests before pushing.

**Comment by afiskon** [comment]
@mkindahl Oops, there was automerge enabled 🤷‍♂️ I will check the `infinity` and `null` cases and submit follow-up pull requests if needed.

**Comment by mkindahl** [comment]
> @mkindahl Oops, there was automerge enabled man_shrugging I will check the `infinity` and `null` cases and submit follow-up pull requests if needed.

Thanks.

**Pull request merged**